src/anemoi/datasets/data/stores.py

```diff
@@ -111,6 +111,9 @@ def open_zarr(path, dont_fail=False, cache=None):
 
         if DEBUG_ZARR_LOADING:
             if isinstance(store, str):
+                import os
+                if not os.path.isdir(store):
+                    raise NotImplementedError("DEBUG_ZARR_LOADING is only implemented for DirectoryStore. Please disable it for other backends.")
```

Comment:
Alternatively, one could do
```suggestion
            import os
            if isinstance(store, str) and os.path.isdir(store):
```